influent-java/src/main/java/influent/forward/NioForwardConnection.java

```diff
@@ -93,15 +116,38 @@
                        final long chunkSizeLimit,
                        final int sendBufferSize,
                        final boolean keepAliveEnabled,
-                       final boolean tcpNoDelayEnabled) {
+                       final boolean tcpNoDelayEnabled,
+                       final ForwardSecurity security) {
     this(
         new NioTcpChannel(socketChannel, sendBufferSize, keepAliveEnabled, tcpNoDelayEnabled),
         eventLoop,
         callback,
-        chunkSizeLimit
+        chunkSizeLimit,
+        security
     );
+    this.keepAliveEnabled = keepAliveEnabled;
 
-    channel.register(eventLoop, SelectionKey.OP_READ, this);
+    if (this.security.isEnabled()) {
+      state = ConnectionState.INIT;
+      try {
+        // SecureRandom secureRandom = SecureRandom.getInstanceStrong();
+        // Above secureRandom may block...
+        SecureRandom secureRandom = SecureRandom.getInstance("NativePRNGNonBlocking");
```

Comment:
SecureRandom can be reused.
And it may be good to store it to ThreadLocal, so I will examine how to do.
Would you add the comment "TODO: reuse SecureRandom"?

Reply:
OK, I will add TODO comment.